apps/aecore/test/aecore_pof_SUITE.erl

```diff
@@ -0,0 +1,262 @@
+
+    %% Add a transaction so there will be a micro block with fraud produced
+    {ok, _Tx3} = add_spend_tx(N1, 1000, 1,  3,  10),
+    {ok, [_Key3, Micro3 | _]} =  aecore_suite_utils:mine_micro_blocks(N2, 1),
+    ct:log("Micro3: ~p", [Micro3]),
+
+    %% Check that the fraud would not be reported again.
+    {ok, Micro4, _} =  rpc:call(N2, aec_block_micro_candidate, create, [Micro3]),
+    ?assertEqual(no_fraud, aec_blocks:pof(Micro4)),
+
+    Header1 = aec_blocks:to_header(Micro1),
+    Header2 = aec_blocks:to_header(Micro2S),
+
+    case aec_blocks:pof(Micro3) of
+        no_fraud -> error({expected_fraud, Micro3});
+        PoF ->
+            ?assertEqual(lists:sort([Header1, Header2]),
+                         lists:sort([aec_pof:header1(PoF), aec_pof:header2(PoF)])),
+            ?assertEqual(rpc:call(N1, aec_keys, pubkey, []),
+                         {ok, aec_pof:pubkey(PoF)})
+    end,
+
+    check_post(N1, N2),
+
+    ok.
+
+siblings_on_micro_block(Config) ->
+    MBC = ?config(micro_block_cycle, Config),
+    aecore_suite_utils:start_node(dev1, Config),
+    aecore_suite_utils:start_node(dev2, Config),
+
+    N1 = aecore_suite_utils:node_name(dev1),
+    N2 = aecore_suite_utils:node_name(dev2),
+
+    aecore_suite_utils:connect(N1),
+    aecore_suite_utils:connect(N2),
+
+    aecore_suite_utils:mine_key_blocks(N1, 1),
+
+    Account1 = #{ pubkey := PK1 } = new_keypair(),
+    Account2 = #{ pubkey := PK2 } = new_keypair(),
+
+    {ok, Tx0a} = add_spend_tx(N1, 100000, 1, 1, 10, patron(), PK1),
+    {ok, Tx0b} = add_spend_tx(N1, 100000, 1, 2, 10, patron(), PK2),
+
+    aecore_suite_utils:mine_blocks_until_txs_on_chain(N1, [Tx0a, Tx0b], 5),
+
+    {ok, _Tx1} = add_spend_tx(N1, 1000, 1,  3, 10),
+
+    {ok, [_Key, Micro0 | _]} = aecore_suite_utils:mine_micro_blocks(N1, 1),
+
+    %% Act fraudulent on "behalf" of N1
+    timer:sleep(MBC), %% adhere to micro block interval
+    {ok, _Tx2} = add_spend_tx(N1, 1000, 1,  1,  10, Account1, new_pubkey()),
+    {ok, Micro1, _} =  rpc:call(N1, aec_block_micro_candidate, create, [Micro0]),
+    {ok, Micro1S} = rpc:call(N1, aec_keys, sign_micro_block, [Micro1]),
+    timer:sleep(MBC), %% adhere to micro block interval
+    {ok, _Tx3} = add_spend_tx(N1, 1000, 1,  1,  10, Account2, new_pubkey()),
+    {ok, Micro2, _} =  rpc:call(N1, aec_block_micro_candidate, create, [Micro0]),
+    {ok, Micro2S} = rpc:call(N1, aec_keys, sign_micro_block, [Micro2]),
+
+    rpc:call(N2, aec_conductor, post_block, [Micro1S]),
+    rpc:call(N2, aec_conductor, post_block, [Micro2S]),
+
+    %% Add a transaction so there will be a micro block with fraud produced
+    {ok, _Tx4} = add_spend_tx(N1, 1000, 1,  4, 10),
+    {ok, [_Key3, Micro3|_]} =  aecore_suite_utils:mine_micro_blocks(N2, 1),
+    ct:log("Micro3: ~p", [Micro3]),
+
+    %% Check that the fraud would not be reported again.
+    {ok, Micro4, _} =  rpc:call(N2, aec_block_micro_candidate, create, [Micro3]),
```

Comment:
Update is not affected, nor tested. We could imagine looking for fraud there as well, but I made the decision that since the restriction of reporting in the first micro block was lifted, doing this in `aec_block_micro_candidate:update` is not important.